ros/src/twist_controller/twist_controller_mod.py

```diff
@@ -32,60 +32,55 @@ def __init__(self, default_update_interval, wheel_base, steer_ratio, min_speed,
     def handle_dynamic_variable_update(self, config, level):
         # reset PID controller to use new parameters
         self.setup_pid_controllers(config['dyn_velo_proportional_control'], config[
-            'dyn_accel_proportional_control'], config['dyn_accel_integral_control'], self.max_deceleration, self.max_acceleration)
-
-        # reset acceleration cruise constant
-        self.acceleration_cruise_constant = config['dyn_acceleration_cruise_constant']
+            'dyn_velo_integral_control'], config['dyn_braking_proportional_control'], config['dyn_braking_integral_control'])
 
         return config
 
-    def setup_pid_controllers(self, velo_p, accel_p, accel_i, max_deceleration, max_acceleration):
-        rospy.loginfo("Initializing PID controllers with velo_P: {}, accel_P: {}, accel_I: {}"
-                      .format(velo_p, accel_p, accel_i))
+    def setup_pid_controllers(self, velo_p, velo_i, braking_p, braking_i):
+        rospy.loginfo("Initializing PID controllers with velo_P: {}, velo_I: {}, braking_P: {}, braking_I: {}"
+                      .format(velo_p, velo_i, braking_p, braking_i))
         # create velocity pid controller thresholded between min and max
         # acceleration values
         self.velocity_pid_controller = PID(
-            velo_p, 0, 0, max_deceleration, max_acceleration)
+            velo_p, velo_i, 0, 0, 1)
 
         # create acceleration pid controller thresholded between 0% and 100%
         # for throttle
-        self.acceleration_pid_controller = PID(accel_p, accel_i, 0.0, 0.0, 1.0)
+        self.braking_pid_controller = PID(
+            braking_p, braking_i, 0.0, 0.0, 10000)
 
     def control(self, target_linear_velocity, target_angular_velocity, current_linear_velocity):
         # compute timestep
         timestep = self.compute_timestep()
-        # use velocity pid controller to compute desired acceleration
-        acceleration_command = self.compute_acceleration_command(
-            target_linear_velocity - current_linear_velocity, timestep)
-        rospy.logwarn('Current linear velocity %.2f, target linear velocity %.2f, desired total acceleration %.2f',
-                      current_linear_velocity, target_linear_velocity, acceleration_command)
+        velocity_error = target_linear_velocity - current_linear_velocity
+
+        if (target_angular_velocity == 0 and current_linear_velocity == 0):
```

Comment:
Should this be "target_linear_velocity"?

Reply:
Yes! Nice catch.